spec/build_spec.lua

```diff
@@ -451,3 +451,447 @@ describe("LuaRocks build tests #integration", function()
+            
+            rockspecs.from_persisted_table("test-1.0-1.rockspec", rockspec)
+            assert.truthy(build_builtin.run(rockspec))
+            assert.truthy(lfs.attributes("lib/luarocks/rocks-" .. test_env.lua_version .. "/test/1.0-1/lua/test.lua"))
+         end)
+         
+         it("automatically extracts the modules and libraries if they are not given and builds against any external dependencies", function()
+            local ssllib = "ssl"
+            if test_env.TEST_TARGET_OS == "windows" then
+               if test_env.MINGW then
+                  ssllib = "eay32"
+               else
+                  ssllib = "ssleay32"
+               end
+            end
+
+            local rockspec = {
+               rockspec_format = "3.0",
+               package = "c_module",
+               version = "1.0-1",
+               source = {
+                  url = "http://example.com/c_module"
+               },
+               external_dependencies = {
+                  openssl = {
+                     library = ssllib -- Use OpenSSL since it is available on all testing platforms
+                  }
+               },
+               build = {
+                  type = "builtin"
+               }
+            }
+            write_file("c_module.c", c_module_source, finally)
+            
+            rockspecs.from_persisted_table("c_module-1.0-1.rockspec", rockspec)
+            if test_env.TEST_TARGET_OS == "windows" then
+               local mingw_dir = ""
+               if test_env.MINGW then
+                  mingw_dir = "\\MinGW"
+               end
+               
+               rockspec.variables["openssl_INCDIR"] = "C:\\OpenSSL-Win32\\include" .. mingw_dir
+               rockspec.variables["openssl_LIBDIR"] = "C:\\OpenSSL-Win32\\lib" .. mingw_dir
+            else
+               rockspec.variables["openssl_INCDIR"] = "/usr/local/opt/openssl/include"
+               rockspec.variables["openssl_LIBDIR"] = "/usr/local/opt/openssl/lib"
+            end
+            assert.truthy(build_builtin.run(rockspec))
+         end)
+         
+         it("returns false if any external dependency is missing", function()
+            local rockspec = {
+               rockspec_format = "3.0",
+               package = "c_module",
+               version = "1.0-1",
+               source = {
+                  url = "https://example.com/c_module"
+               },
+               external_dependencies = {
+                  ext_dep = {
+                     library = "missing"
+                  }
+               },
+               build = {
+                  type = "builtin"
+               }
+            }
+            write_file("c_module.c", c_module_source, finally)
+
+            rockspecs.from_persisted_table("c_module-1.0-1.rockspec", rockspec)
+            rockspec.variables["missing_INCDIR"] = lfs.currentdir()
+            rockspec.variables["missing_LIBDIR"] = lfs.currentdir()
```

Comment:
the variable names that are generated from that `external_dependencies` entry would be `ext_dep_LIBDIR` and `ext_dep_INCDIR`.